dbms/src/Storages/KVStore/KVStore.cpp

```diff
@@ -495,8 +503,13 @@ RegionPtr KVStore::genRegionPtr(metapb::Region && region, UInt64 peer_id, UInt64
         }
         RegionMeta(std::move(peer), std::move(region), std::move(apply_state));
     });
-
-    return std::make_shared<Region>(std::move(meta), proxy_helper);
+    auto new_region = std::make_shared<Region>(std::move(meta), proxy_helper);
+    if (register_to_table)
+    {
+        auto & region_table = tmt.getRegionTable();
+        region_table.addPrehandlingRegion(*new_region);
+    }
+    return new_region;
```

Comment:
`TMTContext` is a complicated class. Passing the TMTContext to `KVStore::genRegionPtr` and register it to region_table will make this method hard to understand.

Maybe add a method like 
```
TMTContext::genRegionPtr(
    metapb::Region && region,
    UInt64 peer_id,
    UInt64 index,
    UInt64 term)
{
    // ...
    auto region = kvstore->genRegionPtr(...);
    region_table.addPrehandlingRegion(*region);
    return region;
}
```

Reply:
Yes, this is a problem, and is confusing.

However, because `TMTContext` is really a complicated class, so IMO it is also kind of strange to me if we create a Region in `TMTContext`. It seems to me a better idea to make RegionTable into KVStore, but it cost some refactoring.

So in general, I think it is better to use `std::optional` over `RegionTable`. Once given, then we will handle the `region_table`.